### ISSUE TITLE
Slow Performance when scrolling a listview with link beyond 30+ items

### ISSUE THREAD

**Issue by rycfung** [watusi/jquery-mobile-iscrollview#25]
Tried using this plugin for a listview with 30+ items, each being a link. The scroll pretty much lags with every swipe and only response only 1-2 seconds later and still lags while it scrolls.

This apparently isn't a problem when I remove a link. I'm using jQuery Mobile 1.1, so I suspect the piano effect shouldn't be the cause here.

**Comment by jtara** [comment]
Are you using the patched jQuery Mobile 1.1 that is provided in the demo directory, the distribution version, or some other version?

There are two serious listview performance bugs in JQM 1.1, both of which are slated to be fixed in 1.1.1. (I think both fixes are already in the repo.) Both of them involve longer listviews with links. I've provided a JQM 1.1 with one of the bugs patched - the other one is still present.

If they don't release 1.1.1 soon, I'll do a patched version of JQM with both fixes.

Have you tried it with 1.0.1/1.6.4 on the same device? What device?

**Comment by rycfung** [comment]
I'm using jqm 1.1.0/1.7.2 with 1.2pre1. I'm testing this on iPhone 4s 5.0.1

Actually I managed to get the scrolling to reasonable speed by adding the following css from demo.css in one of your examples. Are these necessary in order for the scrolling to perform reasonably? 

```
body {
    overflow:hidden;
    -webkit-font-smoothing: subpixel-antialiased;

    -webkit-touch-callout: none;
    -webkit-user-select: none;
    -khtml-user-select: none;
    -ms-user-select: none;
    -moz-user-select: none;
    user-select: none;
}

.ui-navbar a {
    height: 47px;    /* + 2px margins = 49px */
}

.ui-navbar .ui-btn-inner {
    font-size: 14px;
}

@media only screen and (max-height: 280px){
    .ui-header {
        height: 30px;
    }

    .ui-header .ui-title {
        font-size: 14px;
    }


    .ui-header .ui-btn {
        display: none;
    }

    .ui-navbar a {
        height: 37px;    /* + 2px border = 39px */
    }

    .ui-navbar .ui-btn-inner {
        font-size: 13px;
    }
}

@media DISABLED {
    .iscroll-scroller,
    .iscroll-scroller * {
        -webkit-transform: translateZ(0);
    }
}
@media all {
    .iscroll-content li  {
        -webkit-transform: translateZ(0);
    }
}


.iscroll-content .ui-listview li.ui-btn-up-f
{
    background-image: none;
    text-shadow: none;
}

.iscroll-content .ui-listview li.ui-btn-hover-f
{
    background-image: none;
    text-shadow: none;
}

.iscroll-content .ui-listview li.ui-btn-down-f
{
    background-image: none;
    text-shadow: none;
}

.iscroll-content .ui-listview *.ui-btn-up-c a.ui-link-inherit
{
}

.iscroll-content .ui-listview *.ui-btn-hover-c a.ui-link-inherit
{
}

.iscroll-content .ui-listview *.ui-btn-down-c a.ui-link-inherit
{
}

/* Remove background image from non-button scroller content */
.iscroll-content .ui-body-c
{
    background-image: none;
}
```

**Comment by jtara** [comment]
Yes, _some_ of that is needed to get decent performance. It's discussed in the documentation.

```
@media all {
    .iscroll-content li  {
        -webkit-transform: translateZ(0);
    }
}
```

This one is a trade-off. It will avoid jerky scrolling on iOS devices, but may increase initial construction time. If this is left out, scrolling will be jerky until you first reach the bottom of the view.

All the the `.iscroll-content` rules are useful, because background images and text shadows are very costly to render, and so in the demo I have eliminated them from the scrolled content. This is probably more useful on Android than on iOS, as Android seems to have a particular problem with these.

**Comment by jtara** [comment]
I've updated the patched jQuery Mobile 1.1 with the latest patch from the JQM repo, which is probably what will go into JQM 1.1.1. Earlier versions has a preliminary patch for the "long listview" issues.
